### ISSUE TITLE
Cleanup of BaseProvider methods and attributes

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#2291]
In the `BaseProvider` class, there are several things that can be cleaned up:

* `cimrepository` attribute: This is currently simply a public attribute, which is writeable. It could be changed to become a read-only property, with documentation.
* `find_instance()` method: It could be named more consistently as `get_instance()`.
* `find_interop_namespace()` method: It could be named more consistently as `get_interop_namespace()`.
* `get_instance_store()` method: This is purely a wrapper function around `cimrepository.get_instance_store()` and can be removed, if we access the repository through the `cimrepository` property.
* `get_qualifier_store()`: same here

If we do this, it would need to be done in beta, since it changes names.

DISCUSSION

**Comment by KSchopmeyer** [comment]
1. cimrepository is a readonly  property.  See line 254 of _wbemconnectionmock.py

2. find_instance - I did this to avoid confusion with get_instance since that is a specific set of attributes and is tied to the GetInstance (the internal equivalent).  It uses the instance_store as the locator and only gets the instance without filtering it. Note that it is also a static method.  I use it as a shorter path to get instances when we are not passing them on, just using the data internally.

TODO: Karl review differences and reason for both methods.

3. find_interop_namespace().  That one could go either way find or get.   I have no opinion

4.  I kept these wrappers get_..._store() to make life simpler for the provider writer.  My goal was to have ( methods in FakedWbemConnect  but with the existence of BaseProvider, maybe that has changed.

KS TODO review

I agree that it would have to be done in beta,.
